Xamarin.Essentials/FileSystem/FileSystem.android.cs

```diff
@@ -53,6 +54,17 @@ static Task<Stream> PlatformOpenAppPackageFileAsync(string filename)
             }
         }
 
+        // Android does't have the concept for returning Directories (AFICT).
+        // As such, we only return string with no . in them. Not ideal but
+        // should work in most cases as Folders rarely contain dots
+        static string[] PlatformGetAppResourceDirectories(string path)
+            => Platform.AppContext.Assets.List(path).Where(file => !file.Contains(".")).Select(x => Path.Combine(path, x)).ToArray();
```

Comment:
Can we wrap this for easier reading?

Reply:
Android is dodgy and it seems if I pass in `/` then I might get the root of the APK or something based on SO. I tested with Android 30 and I get an empty result. Which is wrong. If you use `""`, then you get the root. Might be worth adding some checks to first make sure that the path coming in is translated to the Android ways.

However, Android is bananas (probably fitting since we support the mono monkey) because Android adds junk:

```
> Platform.AppContext.Assets.List("");
{string[4]}
	[0]: "FileSystemTemplate.txt"
	[1]: "Folder"
	[2]: "images"
	[3]: "webkit"
```

This guy person reports that is also has a `sounds` dir which I don't have: https://stackoverflow.com/a/8366081 However, we cannot filter any of them out since that really is a valid folder and I can add things to it... I can't see the folder in the APK, but I can add things to it and then it exists. Bonkers. It must be a pseudo file structure that is merged with the real APK.

Reply:
I also don't think we can use the filter of "contains period" because I can create an extensionless file. Maybe if we switch to EnumerateFiles, we can also do a `Assets.List("<filename>")` and see if it returns anything:

```
var dirs = Assets.List(path);
foreach (var dir in dirs) 
    if (Assets.List(dir).Length > 0)
        yield return dir
    else
        // this is a file
```

And we can do the reverse for the EnumFiles